pkg/auth/dependency/principal/password/normalize.go

```diff
@@ -0,0 +1,135 @@
+package password
+
+import (
+	"strings"
+
+	"golang.org/x/net/idna"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/unicode/norm"
+
+	"github.com/skygeario/skygear-server/pkg/core/auth/metadata"
+	"github.com/skygeario/skygear-server/pkg/core/config"
+)
+
+type LoginIDNormalizar interface {
+	Normalize(loginID string) (string, error)
+	GenerateUniqueKey(normalizeLoginID string) (string, error)
```

Comment:
I think `ComputeUniqueKey` would be a better name. `Generate` seems it'll be different for each call.